Shared/Samples/Show viewshed from point on map/ShowViewshedFromPointOnMapView.swift

```diff
@@ -0,0 +1,183 @@
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct ShowViewshedFromPointOnMapView: View {
+    /// The error shown in the error alert.
+    @State private var error: Error?
+    /// The point on the screen where the user tapped.
+    @State private var tapScreenPoint: Point?
+    /// The graphics overlay that shows where the user tapped on the screen.
+    @State private var inputGraphicsOverlay = {
```

Comment:
Currently, there are 5 ArcGIS types which don't emit state changes that got held as a state property - inputGraphicsOverlay, resultGraphicsOverlay, geoprocessingTask, geoprocessingJob, and map.

While it is OK to hold them as state properties, my personal preference would be - if there are more than 3 types that need to be held, let's create a data model for them. i.e.,

```swift
import ArcGIS
import SwiftUI

struct ShowViewshedFromPointOnMapView: View {
    /// The model used to store the geo model and other expensive objects
    /// used in this view.
    private class Model: ObservableObject {
        /// A map with topographic basemap.
        let map = Map(basemapStyle: .arcGISTopographicBase)

        // …
    }
    
    /// The data model for the sample.
    @StateObject private var model = Model()
    
    var body: some View {
        MapView(map: model.map)
        // …
    }
}
```

Data manipulating methods can also be moved to the data model.

If the data model is even longer, feel free to put it in an extension to the view, like in other samples.